src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
+        T oldValue = oldNode.item;
+        oldNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexOfPresentElement(index);
+        return unlink(findNode(index));
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (node.item == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (object.equals(node.item)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> findNode(int index) {
+        if (index < size / 2) {
+            Node<T> node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+            return node;
+        } else {
+            Node<T> node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+            return node;
+        }
+    }
+
+    private void linkFirst(T element) {
+        final Node<T> first = head;
+        final Node<T> newNode = new Node<>(null, element, first);
+        head = newNode;
+        if (first == null) {
+            tail = newNode;
+        } else {
+            first.prev = newNode;
+        }
+        size++;
+    }
+
+    private void linkLast(T element) {
+        final Node<T> last = tail;
+        final Node<T> newNode = new Node<>(last, element, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
+    }
+
+    private void linkBefore(T element, Node<T> node) {
+        final Node<T> current = node.prev;
+        final Node<T> newNode = new Node<>(current, element, node);
+        node.prev = newNode;
+        if (current == null) {
+            head = newNode;
+        } else {
+            current.next = newNode;
+        }
+        size++;
+    }
+
+    private T unlink(Node<T> node) {
+        final T element = node.item;
+        final Node<T> next = node.next;
+        final Node<T> prev = node.prev;
+
+        if (prev == null) {
+            head = next;
+        } else {
+            prev.next = next;
+            node.prev = null;
+        }
+
+        if (next == null) {
+            tail = prev;
+        } else {
+            next.prev = prev;
+            node.next = null;
+        }
+
+        node.item = null;
+        size--;
+        return element;
```

Comment:
Create a separate method `unlink(Node node)`. This logic will be used in two methods: when we remove from the list by index and by value. As a result, we will call it from both methods when Node is found.